share/eds/retriever_test.go

```diff
@@ -124,3 +129,66 @@ func TestRetriever_MultipleRandQuadrants(t *testing.T) {
 	_, err = ses.Reconstruct(ctx)
 	assert.NoError(t, err)
 }
+
+func TestFraudProofValidation(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer t.Cleanup(cancel)
+	bServ := mdutils.Bserv()
+
+	odsSize := []int{2, 4, 16, 32, 64, 128}
+	for _, size := range odsSize {
+		t.Run(fmt.Sprintf("ods size:%d", size), func(t *testing.T) {
+			var errByz *byzantine.ErrByzantine
+			faultHeader, err := generateByzantineError(ctx, t, 32, bServ)
+			require.True(t, errors.As(err, &errByz))
+
+			p := byzantine.CreateBadEncodingProof([]byte("hash"), uint64(faultHeader.Height()), errByz)
+			err = p.Validate(faultHeader)
+			require.NoError(t, err)
+		})
+	}
+}
+
+func generateByzantineError(
+	ctx context.Context,
+	t *testing.T,
+	odsSize int,
+	bServ blockservice.BlockService,
+) (*header.ExtendedHeader, error) {
+	eds := edstest.RandByzantineEDS(t, odsSize)
+	err := ipld.ImportEDS(ctx, eds, bServ)
+	require.NoError(t, err)
+	h := headertest.ExtendedHeaderFromEDS(t, 1, eds)
+	_, err = NewRetriever(bServ).Retrieve(ctx, h.DAH)
+
+	return h, err
+}
+
+func BenchmarkBEFP(b *testing.B) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer b.Cleanup(cancel)
+	bServ := mdutils.Bserv()
+
+	odsSize := []int{2, 4, 16, 32, 64, 128}
+	for _, size := range odsSize {
+		b.Run(fmt.Sprintf("ods size:%d", size), func(b *testing.B) {
+			t := &testing.T{}
+			eds := edstest.RandByzantineEDS(t, size)
+			err := ipld.ImportEDS(ctx, eds, bServ)
+			require.NoError(t, err)
+			h := headertest.ExtendedHeaderFromEDS(t, 1, eds)
+			r := NewRetriever(bServ)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				b.ReportAllocs()
+				_, err = r.Retrieve(ctx, h.DAH)
+				var errByz *byzantine.ErrByzantine
+				require.ErrorAs(t, err, &errByz)
```

Comment:
Retrieval operation is very long and polluting benchmark results. Lets extract `byzantine.NewErrByzantine` into separate benchmark. And then convert this benchmark to only observe time of validation:
```
p := byzantine.CreateBadEncodingProof([]byte("hash"), uint64(h.Height()), errByz)
err = p.Validate(h)

Reply:
done